mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -79,6 +79,7 @@ private[ml] trait PredictorParams extends Params
 
     dataset.select(col($(labelCol)).cast(DoubleType), w, col($(featuresCol))).rdd.map {
       case Row(label: Double, weight: Double, features: Vector) =>
+        require (weight >= 0.0, "illegal weight value: " + weight + " weight must be >= 0.0")
```

Comment:
I was going to wonder if we should require weights to be positive, but I don't think we do now. I guess we tend to catch it when we check whether all weights are 0 (sum = 0).
BTW you can use string interpolation in these error messages.